### ISSUE TITLE
Firefox can't set password for users

### ISSUE THREAD

**Issue by lachlan-00** [jellyfin/jellyfin#522]
when i try to add a user password i just get an auth failure.
![image](https://user-images.githubusercontent.com/1305249/50866221-d354aa80-13f4-11e9-985c-67fd5994755b.png)

I haven't been able to set a password for any users since the intitial jellyfin release in the web interface

**Comment by lachlan-00** [comment]
Hey thanks, I use firefox exclusively so i'll try that now.

**Comment by lachlan-00** [comment]
Confirming that it works outside firefox in chrome

**Comment by drago-96** [comment]
Which version of Firefox and Jellyfin are you on?
I am on Firefox 64 and Jellyfin 10.0.1 and everything seems to work...
Which steps lead you to that error?

**Comment by Rudy-MS** [comment]
I am using Firefox Quantum 64.0 with a version of Jellyfin of 10.0.1.0. OS: Ubuntu 18.10 Server: Debian

I just tried changing my password with Firefox Q 64.0 and it worked.......just once.....which is odd since 12 hours ago i couldn't change my password with FF even once at all. Tried again with FF to see if I can have success twice but to no avail. I got the "Invalied username or password. Please try again."

Opened up Opera 57(the newest one i believe) and it work just fine...multiple times. So.....

UPDATE: Firefox Quantum 64.0 will work MAYBE once with a user change password attempt. After that it will give the invalid user message. Proceed to Opera and it will work just fine...?     ¯\_(⊙︿⊙)_/¯

**Comment by JustAMan** [comment]
Can you check using Incognito tab? This might be due to some caching thing...

**Comment by joshuaboniface** [comment]
I think it's a caching issue - I got this when trying to add a user to a previously-Emby upgraded box on FFQ 60 and JF 10.0.0. Cleared cache, restarted browser, and suddenly it worked.

**Comment by sparky8251** [comment]
Going to echo these two fine folks sentiment. Tested and unable to reproduce. The caching of web elements is *super* aggressive.

Clear the cache and try again or open the F12 menu in FF, switch to network, tick the disable cache box, and refresh. Only after the refresh completes do you want to close the F12 menu.

**Comment by Rudy-MS** [comment]
Success! @JustAMan I went into incognito mode with FF changed a user profiles password multiple times with no problem. Is this an FF problem or will something need to change in Jellyfin?

**Comment by JustAMan** [comment]
@Rudy-MS I think it's Jellyfin server part trying to be too smart with cache, but I'm not sure.

**Comment by sparky8251** [comment]
This issue can be closed right? We found 80 and 365 day caching period stuff in the backend and so far, clearing the cache has fixed this for users.

**Comment by Rudy-MS** [comment]
@hawken93 I concur with hawken93. This is happening to me on a pretty regular basis. Old and new installs of Jellyfin. Please do not close until fixed. Also please let me know if there is anything I can do for testing and troubleshooting to help you guys fix the code. Thank you!

**Comment by sparky8251** [comment]
So, just to add fuel to this mystery:

Got a friend who is on Emby 3.5.3 and just informed me that he is experiencing these *exact* symptoms. Down to a cache clear in FF helping and Chrome not having this issue.

So, this is a confirmed inherited bug in my eyes rather than a regression. Wonder why its so inconsistent and how we can force it to happen?

**Comment by JustAMan** [comment]
Could this be relevant to why we're seeing this? https://stackoverflow.com/a/14842553
Note that the response says that Chrome is no longer doing force cache invalidation on some cases, but it still shows that some browsers treat cache thing differently. Worth more investigation I think as soon as we have a stable reproducer.

**Comment by sparky8251** [comment]
Again with chrome doing non-standard shit. They do this with HTTP/1.1 requests too. 

They do some whacko parallelization of requests even though the protocol doesnt support it. Seen that cause bugs because devs then make dumb assumptions on how things work.

**Comment by hawken93** [comment]
```
jellyfin[29625]: [17:08:51] [INF] HTTP GET http://jellyfin:8096/Users/<guid-testqwe>. UserAgent: UA
jellyfin[29625]: [17:08:51] [INF] HTTP Response 200 to 10.159.2.242. Time: 1ms. http://jellyfin:8096/Users/<guid-testqwe>
jellyfin[29625]: [17:08:57] [INF] HTTP POST http://jellyfin:8096/Users/<guid-testqwe>/Password. UserAgent: UA
jellyfin[29625]: [17:08:57] [ERR] Error authenticating with provider Default
jellyfin[29625]: System.Exception: Invalid username or password
jellyfin[29625]:    at Emby.Server.Implementations.Library.DefaultAuthenticationProvider.Authenticate(String username, String password, User resolvedUser) in /root/jellyfin/Emby.Server.Implementations/Library/DefaultAuthenticationProvider.cs:line 39
jellyfin[29625]:    at Emby.Server.Implementations.Library.UserManager.AuthenticateWithProvider(IAuthenticationProvider provider, String username, String password, User resolvedUser) in /root/jellyfin/Emby.Server.Implementations/Library/UserManager.cs:line 383
jellyfin[29625]: [17:08:57] [INF] Authentication request for testqwe has been denied.
jellyfin[29625]: [17:08:57] [ERR] Error processing request
jellyfin[29625]: System.ArgumentException: Invalid user or password entered.
jellyfin[29625]:    at MediaBrowser.Api.UserService.PostAsync(UpdateUserPassword request) in /root/jellyfin/MediaBrowser.Api/UserService.cs:line 433
jellyfin[29625]:    at Emby.Server.Implementations.Services.ServiceExecGeneral.GetTaskResult(Task task) in /root/jellyfin/Emby.Server.Implementations/Services/ServiceExec.cs:line 119
jellyfin[29625]:    at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost appHost, IRequest httpReq, IResponse httpRes, ILogger logger, String operationName, CancellationToken cancellationToken) in /root/jellyfin/Emby.Server.Implementations/Services/ServiceHandler.cs:line 96
jellyfin[29625]:    at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken) in /root/jellyfin/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs:line 654
jellyfin[29625]: [17:08:57] [INF] HTTP Response 400 to 10.159.2.242. Time: 19ms. http://jellyfin:8096/Users/<guid-testqwe>/Password
```

**Comment by hawken93** [comment]
```
Parameters:
  CurrentPw | oSMWepfNAb0oeFkpvicV
  NewPw | asdf
```

**Comment by hawken93** [comment]
OK, so don't do that folks ^ I just posted my password on the internet. No worries, I changed it now but this also means that it tries to change the password for the new user from the current users password. Sigh, it's probably firefox automatically filling in an invisible password field.

**Comment by hawken93** [comment]
The moral is that the frontend should be smart enough ti not do this. I see it as a frontend issue

**Comment by hawken93** [comment]
A quick fix is to make it a hidden **text** field. And take care not to make it as a password field to change it later, then I fear the browser could possibly already have filled it.

**Comment by hawken93** [comment]
Alternately, change it into a text field, hide and empty it

**Comment by hawken93** [comment]
https://cloudfour.com/thinks/autofill-what-web-devs-should-know-but-dont/

**Comment by hawken93** [comment]
Tricky because if you actually are changing your own password, you want autofill. Otherwise no

**Comment by hawken93** [comment]
To be honest I think the whole thing should be redone in the most boring way possible so that it feels more secure, fix the default options and make a opt-in box to allow signing in with no password. Fix the salting and so on. I think this covers like 5 different issues we have and would make me feel much better about exposing it on the internet

**Comment by hawken93** [comment]
you can see this by inspecting and unhiding the currentpassword field. if it contains a password, then you know you're in for a bad time. :P

**Comment by hawken93** [comment]
Workaround:
Open inspector, unhide the current password field and empty it

**Comment by lachlan-00** [comment]
I have found that the latest Jellyfin works with firefox. (Jellyfin Web 10.0.1.0)
I did have to ctrl + f5 to get it to save but that didn't work for me previously

@hawken93 could that be if i have saved a password for my main account? that might actually be the issue is that my user account is filling the hidden field and failing.

**Comment by JustAMan** [comment]
@lachlan-00 according to https://github.com/jellyfin/jellyfin-web/pull/7 it is indeed autofill of hidden field that breaks stuff. We expect this to be fixed in next Jellyfin release, as fixing PR is already merged.

**Comment by hawken93** [comment]
@lachlan-00 Yes. When I ctrl+f5, stuff loads a bit slower so I suspect that it's a race condition that usually the javascript clears the field before firefox fills it, but when it loads slower, I think the javascript manages to clear it. :)

**Comment by hawken93** [comment]
Fixed in 10.1.0

**Comment by Bond-009** [comment]
Fixed by jellyfin/jellyfin-web#7
